File: Day_1/Solutions Day 1/TroubleSort.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin>>t;
    for(int T = 0 ;T<t ;T++){
        int n;
        cin>>n;
        int A[n];
        int B[n];
        for(int i = 0 ; i<n ; i++){
            int x;
            cin>>x;
            A[i] = x;
        }
        int cnt_one = 0;
        for(int i = 0 ; i<n ; i++){
            int x;
            cin>>x;
            B[i] = x;
            if(B[i] == 1)cnt_one++;
        }
        bool is_sorted = true;
    /*    A[5] = A[4]
        print(A[6])*/

        for(int i = 0 ; i<n-1 ; i++){
            if(A[i] > A[i+1]){
                is_sorted = false;
                break;
            }
        }
        // 0 0 0 0 , n = 4
        if((cnt_one < n && cnt_one >= 1) || is_sorted){
            cout<<"Yes"<<endl;
        }
        else{
            cout<<"No"<<endl;
        }
    }
}

File: Day_1/Solutions Day 1/TroubleSort.py
T = int(input())
List_of_answers = []
for _ in range(T):
    n = int(input())
    A = list(map(int,input().split()))
    B = list(map(int,input().split()))
    zeros, ones = 0, 0
    for i in B:
        if i:
            ones += 1
        else:
            zeros += 1

    is_sorted = True
    for i in range(0,n-1):
        if A[i] > A[i+1]:
            is_sorted = False
            break

    if (ones>0 and zeros>0) or is_sorted:
        List_of_answers.append("YES")
    else:
        List_of_answers.append("NO")

for ans in List_of_answers:
    print(ans)
